js/civs.js

```diff
@@ -1,18 +1,199 @@
-function civ(name, tree) {
-    resetToDefault(tree);
+function parseCiv(civ) {
+   let builder = new CivBuilder({uniqueUnit: civ.unique[0],
+         uniqueUnitElite: civ.unique[1],
+         uniqueTechOne: civ.unique[2],
+         uniqueTechTwo: civ.unique[3],
+         monkPrefix: civ.monkPrefix});
+   
+   // Disable bldgs, units, techs.
+   if (civ.disabled) {
+      if (civ.disabled.buildings) builder.disableBuildings(civ.disabled.buildings);
+      if (civ.disabled.units) builder.disableUnits(civ.disabled.units);
+      if (civ.disabled.techs) builder.disableTechs(civ.disabled.techs);
+   }
 
-    let selectedCiv = civsConfig[name];
+   // Enable bldgs, units, techs.
+   if (civ.enabled) {
+      if (civ.enabled.buildings) builder.enableBuildings(civ.enabled.buildings);
+      if (civ.enabled.units) builder.enableUnits(civ.enabled.units);
+      if (civ.enabled.techs) builder.enableTechs(civ.enabled.techs);
+   }
 
-    let enabled = selectedCiv.enabled || {};
-    let disabled = selectedCiv.disabled || {};
-    let uniqueConfig = selectedCiv.unique || {};
-    if (selectedCiv.disableHorses) {
-        disableHorses();
-    }
+   if (civ.disableHorses) builder.disableHorses();
+   
+   return builder.build();
+}
+
+function civ(name) {
+   let selectedCiv = parseCiv(civsConfig[name]);
+   const id_regex = /(?<type>.+)_(?<id>([\d]+|unique_unit|elite_unique_unit))_(x|copy)/;
+
+   // On reset, we need to NOT enable things we're about to disable again.
+   SVG.select('.cross').each(function(i) {
+      // If fill opacity is already 0, then this is already enabled.
+      if (this.attr('fill-opacity') === 0) return;
```

Comment:
We could replace this comment with an extracted `isAlreadyEnabled()` function

Reply:
Done.